x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/RuleQueryBuilder.java

```diff
@@ -281,8 +283,11 @@ protected int doHashCode() {
         declareStandardFields(PARSER);
     }
 
-    public static RuleQueryBuilder fromXContent(XContentParser parser) {
+    public static RuleQueryBuilder fromXContent(XContentParser parser, XPackLicenseState licenseState) {
         try {
+            if (QueryRulesConfig.QUERY_RULES_LICENSE_FEATURE.check(licenseState) == false) {
+                throw LicenseUtils.newComplianceException(NAME);
+            }
```

Comment:
Should we throw the exception outside of the try block? We don't want to wrap the license exception under the parsing one.

Reply:
Yeah, I'd originally put it there and was trying to make it prettier. 😅 I'll change it back.